src/course_generator/analysis/JPanelAnalysisSpeedSlope.java

```diff
@@ -477,7 +470,7 @@ public void Refresh(TrackData track, CgSettings settings) {
 				if (btSpeedSlopeFilter.isSelected())
 					s = r.tmp1;
 				else
-					s = r.getSpeed(settings.Unit);
+					s = r.getSpeed();
```

Comment:
We don't need to specify the unit because when we set the speed (see setSpeed() calls) we already specified the unit. That's why we were obtaining different values